File: 20220915/모범답안/20220915_06.js/WordTrain.py
def solution(s):
    """
    :param s: str
    :return: int
    """

    alphabets = {}
    fromIndex = 0
    result = 0

    for i, c in enumerate(s):
        if c in alphabets and fromIndex <= alphabets[c]:
            fromIndex = alphabets[c] + 1

        alphabets[c] = i
        result = max(result, (i - fromIndex) + 1)

    return result

File: 20220915/모범답안/20220915_06.js/WordTrain.js
/**
 * @param s {string}
 * @returns {number}
 */
function solution(s) {
  const alphabets = {};
  let fromIndex = 0;

  return s.split('')
    .reduce((longest, c, i) => {
      if (fromIndex <= alphabets[c]) {
        fromIndex = alphabets[c] + 1
      }

      alphabets[c] = i;
      return Math.max(longest, (i - fromIndex) + 1);
    }, 0);
}

solution